### ISSUE TITLE
Timestamps are parsed as UTC when there is no system-wide tzinfo data

### ISSUE THREAD

**Issue by a-rodin** [timberio/vector#1201]
When there is no `tzinfo` database on the system, three tests related to parsing timestamps fail:

```
---- types::tests::parse_timestamp_auto stdout ----
thread 'types::tests::parse_timestamp_auto' panicked at 'assertion failed: `(left == right)`
  left: `Ok(2001-02-03T11:05:06Z)`,
 right: `Ok(2001-02-03T04:05:06Z)`', src\types.rs:291:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.                                                      
---- types::tests::timestamp_conversion stdout ----
thread 'types::tests::timestamp_conversion' panicked at 'assertion failed: `(left == right)`
  left: `Ok(Timestamp(2001-02-03T11:05:06Z))`,
 right: `Ok(Timestamp(2001-02-03T04:05:06Z))`', src\types.rs:274:9                                                                  
---- types::tests::timestamp_param_conversion stdout ----
thread 'types::tests::timestamp_param_conversion' panicked at 'assertion failed: `(left == right)`
  left: `Ok(Timestamp(2001-02-03T11:05:06Z))`,
 right: `Ok(Timestamp(2001-02-03T04:05:06Z))`', src\types.rs:282:9
```

Maybe we need to bundle timezone info into the Vector binary, although it is has some drawbacks (mainly because timezone info changes quite frequently, and it would require the user to update Vector in case of such changes).

**Comment by binarylogic** [comment]
Yeah, we put a note [here](https://docs.vector.dev/about/data-model/log#fields) under the timestamp type. This was mentioned by @bruceg in https://github.com/timberio/vector/issues/551#issuecomment-513505780.

**Comment by binarylogic** [comment]
@a-rodin should this be part of your project? Is it something you want to address for windows or any of the builds?

**Comment by a-rodin** [comment]
@binarylogic I think that ideally we want to do this for all builds, just to have consistency between all platforms.

**Comment by binarylogic** [comment]
@bruceg this is a pretty old issue, but I'm wondering if this is still relevant? If it's not, feel free to close.

**Comment by a-rodin** [comment]
I'm still somewhat concerned that the behavior is different when timezone files are present on the system and when they are not. Maybe Vector could produce an error in case if a naive timestamp is attempted to be parsed, but no timezone files are present, and the timestamp specifier doesn't specify that it should be parsed as UTC explicitly?

**Comment by bruceg** [comment]
The issue is still relevant, as the potential for problems is still present. There have been two solutions presented:

1. Bundle timezone info into the Vector binary. If it's feasible, it has the problems mentioned previously—the Vector binary would need to be rebuild regularly to support time zone data updates.

2. Raise an error (or at least a warning) when parsing a timestamp with a timezone when no files are present. Hypothetically this could be done at startup, but it might be difficult to solve both reliably and portably.
